### ISSUE TITLE
NodeMCU V3 Lolin instant reset when Rx is connected to Arduino

### ISSUE THREAD

**Issue by Kestranor** [esp8266/Arduino#3536]
Hi!

I have an issue that I'm having a bit of trouble solving. In my application, I'm using a NodeMCU and an Arduino together (which is supposed to drive LEDs and other peripherals) and I'm trying to connect them via HW serial. The communication between the 2 are minimal and event-based. I'm using a speed of 115200 baud. I'm aware of the logic level differences, and I tried testing the connection using a 3.3V Arduino and a 5V arduino using a LLC, in both cases the result is the same.

As soon as the NodeMCU's Rx is connected to the Tx of the arduino (directly in case of the 3v3 Arduino or through the LLC in case of the 5V arduino), the nodeMCU resets and stays in a reset loop until the connection from the Rx pin is removed! After that, it boots and works as intended.

I have also tested both controllers separately, connected to my PC. In that case, serial works perfectly. Both modules are stable and are able to receive and interpret serial commands perfectly sent through the serial monitor. In these cases, the modules are connected throught the USB to the PC.  When the Serial lines are connected together, the USB ports are not used, instead, both are powered from an external supply, which is able to provide enough power (this is 100% not the issue).

Honestly, I don't know if this is the right place to post, as I'm not sure what causes this problem. The NodeMCU's HW serial seems to be working fine when connected to the PC, I just don't understand why it triggers a reset loop as soon as it is connected to the arduino. I don't think it has anything to do with message reception either, since no messages are sent initially, only when certain events happen (buttons press, configuration change, etc). Maybe someone can help. I'd really appreciate it.  Thanks.

**Comment by pieman64** [comment]
Have you tried software serial, the esp variety?

Or perhaps the secondary hardware pins?

**Comment by Kestranor** [comment]
I did not try Software Serial yet, as all the other pins on my NodeMCU are in use and read that it can possibly introduce other problems. But you are right, it is worth a try for debug purposes. I'll see how it goes when I get home.

**Comment by suculent** [comment]
This seems definitely like a hardware issue. I have experienced similar behaviour when having GPIO2/TXD connected to some impedance on power-on… when I connect it after power-on (having it grounded while the ESP gets first juice), everything works great.

How does it behave when you connect the RX after powering the whole thing on?

Maybe try a 10k pull resistor?

M.

**Comment by Kestranor** [comment]
I tried that as well, interestingly enough, as soon as I connect Rx, even after power on, the NodeMCU resets. Messages are not transmitted from the Arduino, it seems to reset immediately. Kind of reminds me of the behavior when sometimes I start the serial monitor and the board resets once... but somehow I don't think this is the same, as in this case, it keeps resetting as long as Rx is connected to the Arduino.

**Comment by Kestranor** [comment]
Update: the issue is solved. It turns out it was the most common type of error: user error. Before I tried SoftwareSerial, I rechecked my arduino code as well as the board itself, and it seems like in my rush to test, I forgot to turn off my debug messages in the version I had uploaded on my Arduino. Of course I turned it off in the code, but apparently I forgot to update the running software before testing. Consequently, the Arduino was probably flooding the ESP with debug messages. Ugh. Silly mistake. Anyway, it works now and the issue can be closed.
